todo/src/InputWithLabel.js

```diff
@@ -0,0 +1,15 @@
+import React, {useEffect, useRef} from "react";
+
+const InputWithLabel = (props) => {
+    const inputRef = useRef();
+    useEffect(() =>
+        {inputRef.current.focus()}, []);
+    return (
+        <>
+        <label htmlFor="todoTitle">{props.children}</label>
+        <input id="todoTitle" type="text" name="title" value={props.todoTitle} onChange={props.handleTitleChange} ref={inputRef}/>
```

Comment:
With the new props passed in, this component will look like this:
```suggestion
        <label htmlFor={id}>{props.children}</label>
        <input id={id} type={props.type}  value={props.value} onChange={props.handleTitleChange} ref={inputRef}/>
```